test/test_todo.py

```diff
@@ -0,0 +1,58 @@
+from flask import url_for
+
+from .conftest import create_package, create_group, create_issue, create_advisory, DEFAULT_GROUP_ID, DEFAULT_ADVISORY_ID, DEFAULT_ISSUE_ID
+from app.model.enum import Publication, Severity, Remote
+from app.model.cve import issue_types
+from app.view.todo import get_todo_data
+
+
+def todo_data(todo_data, expected):
+    for entry, data in todo_data.items():
+        if entry == expected:
+            assert len(data) == 1
+            advisory, _, cvegroup =data[0]
+            assert DEFAULT_ADVISORY_ID == advisory.id
+            assert DEFAULT_GROUP_ID == cvegroup.id
+        else:
+            assert not data
+
+
+@create_package(name='foo', version='1.2.3-4')
+@create_issue(id=DEFAULT_ISSUE_ID, issue_type=issue_types[1], remote=Remote.remote, severity=Severity.low, description='cyber')
+@create_group(id=DEFAULT_GROUP_ID, packages=['foo'], affected='1.2.3-3', fixed='1.2.3-4')
+@create_advisory(publication=Publication.published)
+def test_incomplete_advisories(db, client):
+    todo_data(get_todo_data(), 'incomplete_advisories')
+
+
+@create_package(name='foo', version='1.2.3-4')
+@create_issue(id=DEFAULT_ISSUE_ID, issue_type=issue_types[1], remote=Remote.remote, severity=Severity.low, description='cyber')
+@create_group(id=DEFAULT_GROUP_ID, packages=['foo'], affected='1.2.3-3', fixed='1.2.3-4')
+@create_advisory(publication=Publication.scheduled)
+def test_scheduled_advisories(db, client):
+    todo_data(get_todo_data(), 'scheduled_advisories')
+
+
+@create_package(name='foo', version='1.2.3-4')
+@create_issue(id=DEFAULT_ISSUE_ID, issue_type=issue_types[1], remote=Remote.remote, severity=Severity.low, description='cyber')
+@create_group(id=DEFAULT_GROUP_ID, packages=['foo'], affected='1.2.3-3', fixed='1.2.3-4')
+def test_unhandled_advisories(db, client):
+    data = get_todo_data()['unhandled_advisories']
+    assert data
+    cvegroup = data[0][0]
+    assert cvegroup.id == DEFAULT_GROUP_ID
+
+
+@create_issue(id=DEFAULT_ISSUE_ID, issue_type=issue_types[1])
+@create_group(id=DEFAULT_GROUP_ID, packages=['foo'], affected='1.2.3-3', fixed='1.2.3-4')
+def test_unknown_issues_(db, client):
+    data = get_todo_data()
+    assert data['unknown_issues']
+    assert data['unknown_issues'][0].id == DEFAULT_ISSUE_ID
+
+
+@create_issue(id=DEFAULT_ISSUE_ID, issue_type=issue_types[1])
+def test_orphan_issues_(db, client):
+    data = get_todo_data()
+    assert data['orphan_issues']
+    assert data['orphan_issues'][0].id == DEFAULT_ISSUE_ID
```

Comment:
can you add a test calling the HTML endpoint with everything set up that all test cases/categories show some entries? incomplete/scheduled/unhandled advisories unknown and orphan issues? then check if status code is 200. This way we can ensure there are not serious regressions while displaying the page (no need to check content, just status)

Reply:
That's something I was also planning on doing, but first wanted to handle all the separate cases.